### PR TITLE
add bare-metal-dev pipeline

### PR DESCRIPTION
**Pull request by gaohannk**

### CODE REVIEW

File: concourse/pipelines/dev-image-build.yaml [hunk 1]
```diff
@@ -366,3 +434,53 @@ jobs:
       publish_version: ((.:publish-version))
       wf: "debian/debian_10_worker.publish.json"
       environment: "prod"
+- name: publish-to-staging-rhel-7-metal-dev
```
Comment (zmarano) [inline concourse/pipelines/dev-image-build.yaml:437]: These will overwrite the non-dev publish images since they use the same workflow.

Comment (gaohannk) [reply concourse/pipelines/dev-image-build.yaml:437]: Oh, right. We need to a separate image family.

Comment (gaohannk) [reply concourse/pipelines/dev-image-build.yaml:437]: I create a `rhel-7-metal-dev.publish.json` and `rhel-8-metal-dev.publish.json` in linux_dev

File: concourse/pipelines/dev-image-build.yaml [hunk 2]
```diff
@@ -366,3 +434,53 @@ jobs:
...
+  - load_var: publish-version
+    file: publish-version/version  # produced from generate-version task
+  - task: publish-rhel-7-metal-dev
+    file: guest-test-infra/concourse/tasks/daisy-publish-images.yaml
+    vars:
+      source_gcs_path: "gs://gce-image-archive/rhel"
+      source_version: v((.:source-version))
+      publish_version: ((.:publish-version))
+      wf: "linux_dev/rhel_7_metal_dev.publish.json"
+      environment: "staging"
```
Comment (zmarano) [inline concourse/pipelines/dev-image-build.yaml:461]: won't be passing environment with these workflows.

Comment (gaohannk) [reply concourse/pipelines/dev-image-build.yaml:461]: remove this line

**Review by zmarano — APPROVED** [review]

**Pull request merged**